source/common/upstream/upstream_impl.h

```diff
@@ -637,9 +639,14 @@ class ClusterImplBase : public Cluster, protected Logger::Loggable<Logger::Id::u
   const Network::Address::InstanceConstSharedPtr
   resolveProtoAddress(const envoy::api::v2::core::Address& address);
 
-  static HostVectorConstSharedPtr createHostList(const HostVector& hosts, Host::Health health);
-  static HostsPerLocalityConstSharedPtr createHostLists(const HostsPerLocality& hosts,
-                                                        Host::Health);
+  // Partitions the provided list of hosts into two new lists containing the healthy and degraded
+  // hosts respectively.
+  static std::pair<HostVectorConstSharedPtr, HostVectorConstSharedPtr>
```

Comment:
Is it worth doing the same type tricks you did previously to make it more clear that the first is healthy and the second is degraded? I think this might be worth it in the this case?